torch/_refs/__init__.py

```diff
@@ -456,21 +465,37 @@ def _isfinite(a: TensorLikeType) -> TensorLikeType:
 
 
 def _isinf(a: TensorLikeType) -> TensorLikeType:
-    # TODO Add complex tensor support to remove is_infinite prim
-    # if utils.is_complex_dtype(a):
-    #     return bitwise_or(_isinf(real(a), _isinf(imag(a))
-    # else:
-    #     return bitwise_not(bitwise_or(isnan(a), isfinite(a)))
-    if utils.is_float_dtype(a.dtype) or utils.is_complex_dtype(a.dtype):
-        return prims.is_infinite(a)
-
-    return zeros_like(a, dtype=torch.bool)
+    if utils.is_complex_dtype(a.dtype):
+        return bitwise_or(_isinf(real(a)), _isinf(imag(a)))
+    return bitwise_and(bitwise_not(isfinite(a)), bitwise_not(_isnan(a)))
 
 
 isinf = _make_elementwise_unary_reference(
     _isinf,
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.ALWAYS_BOOL,
-    aten_op=torch.ops.aten.isinf,  # prim/aten name mismatch
+    aten_op=torch.ops.aten.isinf,
+)
+
+
+def _isposinf(a: TensorLikeType) -> TensorLikeType:
+    return bitwise_and(_isinf(a), gt(a, zeros_like(a)))
```

Comment:
`gt(a, 0)` -- if `gt` doesn't accept a scalar then please add a TODO comment here and to the gt ref